packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts

```diff
@@ -182,8 +184,8 @@ export const processProtectedData = async ({
       workerpool: vWorkerpool === ethers.ZeroAddress ? 'any' : vWorkerpool, // if address zero was chosen use any workerpool
       app: vApp,
       dataset: vProtectedData,
-      requester: requester, // public orders + user specific orders
-      isRequesterStrict: useVoucher, // If voucher, we only want user specific orders
+      requester: requester,
+      isRequesterStrict: vVoucherOwner ? false : useVoucher, // if voucherOwner is set, we don't want to be strict as the requester will not have an order for him
```

Comment:
If the voucherOwner is defined and is different from the requester, we set isRequesterStrict to false to allow fetching orders from any requester 🆗

but if the requester is the voucherOwner, we must enable strict mode (set isRequesterStrict to true) to ensure fetching voucher orders

here's a suggestion : 👇

```suggestion
isRequesterStrict:
  vVoucherOwner && vVoucherOwner.toLowerCase() !== requester.toLowerCase()
    ? false
    : useVoucher,
```